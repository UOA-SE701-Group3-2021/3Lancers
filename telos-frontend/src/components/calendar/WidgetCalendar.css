.small-month {
  font-size: 10px;
  font-weight: bold;
}

.big-number {
  --bigTextSize: 36px;
  font-size: var(--bigTextSize);
  height: var(--bigTextSize);
  line-height: var(--bigTextSize);
  font-weight: 600;
  color: var(--iconColour);
}