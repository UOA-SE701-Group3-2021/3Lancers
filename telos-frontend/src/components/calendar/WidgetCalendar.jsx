import moment from 'moment';
import widgetStyles from '../Widget.module.css';
import calendarWidgetStyles from './WidgetCalendar.module.css';

const WidgetTodo = ({ addNewCalendar }) => {
  const today = moment();
  const thisMonth = today.format('MMMM');
  const thisDay = today.format('DD');
  return (
    <div onClick={addNewCalendar} className={`${widgetStyles.widgetIcon} ${widgetStyles.vertical}`}>
      <div className={calendarWidgetStyles.smallMonth}>{thisMonth}</div>
      <div className={widgetStyles.bigText}>{thisDay}</div>
    </div>
  );
};

export default WidgetTodo;
