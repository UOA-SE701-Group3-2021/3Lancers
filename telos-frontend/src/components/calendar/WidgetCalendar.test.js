import { fireEvent, render, cleanup } from '@testing-library/react';
import WidgetCalendar from './WidgetCalendar';

afterEach(cleanup);

test('Testing the WidgetCalendar', () => {
  Date.now = jest.fn(() => new Date('2020-04-01'));

  const { asFragment, container } = render(<WidgetCalendar />);

  expect(asFragment(<WidgetCalendar />)).toMatchSnapshot();

  fireEvent.click(container.firstChild);

  expect(container.firstChild).toHaveClass('widgetIcon vertical');
});
