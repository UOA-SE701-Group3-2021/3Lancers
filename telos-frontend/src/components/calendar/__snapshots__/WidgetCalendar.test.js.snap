// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the WidgetCalendar 1`] = `
<DocumentFragment>
  <div
    class="widgetIcon vertical"
  >
    <div
      class="smallMonth"
    >
      April
    </div>
    <div
      class="bigText"
    >
      08
    </div>
  </div>
</DocumentFragment>
`;
